### DISCUSSION TITLE
slices versus batch

### DISCUSSION THREAD

**Discussion by mdanb** [pyg-team/pytorch_geometric#2318] [Q&A] [answered]
I usually use the `batch` attribute to indicate which graph a node belongs to in a `Dataset`. But I'm reading someone else's code and I noticed that they do it using `slices` so I browsed the documentation and noticed that the [example](https://pytorch-geometric.readthedocs.io/en/latest/notes/create_dataset.html) for creating my own dataset also uses `slices`. Are these just two different ways of doing the same thing or is there a more fundamental difference between them?

**Comment by rusty1s** [comment, marked as answer]
They only slightly differ. I'm not really sure how you implement it via `batch`, but `slices` lets you access individual examples in `O(1)`. Furthermore, since data is only sliced, it will not need to duplicate the output data in memory:
```python
x = dataset.data.x[100:150]  # x holds the same underlying memory as dataset.data.x
```
